src/expressions/functions/builtInFunctions_node.ts

```diff
@@ -123,6 +123,32 @@ const fnData: FunctionDefinitionType = (
 	return atomize(sequence, executionParameters);
 };
 
+const fnGenerateId: FunctionDefinitionType = (
+	dynamicContext,
+	executionParameters,
+	_staticContext,
+	nodeValue,
+) => {
+	let node: NodePointer;
+	if (!nodeValue) {
+		if (!dynamicContext || !dynamicContext.contextItem) {
```

Comment:
You could've used the `contextItemAsFirstArgument` helper, that does this internally, but I'm fine either way.